client/state/shortcodes/actions.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	SHORTCODE_FETCH,
+	SHORTCODE_RECEIVE
+} from 'state/action-types';
+
+export function fetchShortcode( siteId, shortcode ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SHORTCODE_FETCH,
+			siteId,
+			shortcode
+		} );
+
+		return wpcom.undocumented().site( siteId ).shortcodes( {
+			shortcode: shortcode
+		}, ( error, data ) => {
```

Comment:
Had you considered writing this as a Promise chain? You will need to make one minor revision to the [undocumented function](https://github.com/Automattic/wp-calypso/blob/e6eb37a38c8f5041e3cdf285bea03afc8ef989fc/client/lib/wpcom-undocumented/lib/site.js#L122) to `return` the `this.wpcom.req.get` call.